MyHashMap.java

```diff
@@ -0,0 +1,228 @@
+package com.company;
+
+import java.util.Objects;
+
+public class MyHashMap<K, V> implements MyMap<K, V> {
+    private Node[] nodesArr;
+    private double loadFactor = 0.75;
+    private int threshold = 12;
+    private int size = 16;
+    private int capacity;
+
+    MyHashMap() {
+        nodesArr = new MyHashMap.Node[size];
+    }
+
+    public void setLoadFactor(double loadFactor) {
+        this.loadFactor = loadFactor;
+    }
+
+    @Override
+    public V put(K key, V value) {
+        remove(key);
+        capacity++;
+        if (capacity >= threshold) {
+            resize();
+        }
+        if (key == null) {
+            putVal(0, value, key);
+            return value;
+        }
+        putVal(hashIndex(key), value, key);
+        return value;
+    }
+
+    public int hashIndex(K key) {
+        if (key == null) {
+            return 0;
+        }
+        return hashCode(key) & (size - 1);
+    }
+
+    public int hashIndex(K key, int n) {
+        if (key == null) {
+            return 0;
+        }
+        return hashCode(key) & (n - 1);
+    }
+
+    public int hashCode(K key) {
+        return Objects.hashCode(key);
+    }
+
+    private void putVal(int index, V value, K key) {
+        remove(key);
+        if (nodesArr[index] == null) {
+            nodesArr[index] = new Node(hashCode(key), key, value, null);
+        } else {
+            Node currentNode = nodesArr[index];
+            while (currentNode.next != null) {
+                currentNode = currentNode.next;
+            }
+            currentNode.next = new Node(hashCode(key), key, value, null);
+
+        }
+    }
+
+    private void resize() {
+        int size = this.size * 2;
+        Node[] tempNode = new MyHashMap.Node[size];
+        System.arraycopy(nodesArr, 0, tempNode, 0, this.size);
+        this.nodesArr = tempNode;
+        this.size = size;
+        this.threshold = (int) (this.size * loadFactor);
+    }
+
+    @Override
+    public V remove(K key) {
+        if (key == null) {
+            if (nodesArr[0].key == null) {
+                Node currNode = nodesArr[0];
+                nodesArr[0] = currNode.next;
+                return currNode.value;
+            } else {
+                if (nodesArr[0].next != null) {
+                    Node currNode = nodesArr[0].next;
+                    if (currNode.key == null) {
+                        nodesArr[0].next = currNode.next;
+                        return currNode.value;
+                    }
+                    Node lastNode;
+                    while (currNode.key != null) {
+                        lastNode = currNode;
+                        currNode = lastNode.next;
+                        if (nodesArr[0].key == null) {
+                            lastNode.next = currNode.next;
+                            currNode = currNode.next;
+                        }
+                    }
+                }
+            }
+
+        } else {
+            int sizeArr = 16;

```

Comment:
Why 16?

Reply:
> Why 16?

the default value in the Java class, rumor has it that this is a good ratio along from the load factor of 0.75

Reply:
this is default value for initial size of array. Why do you need sizeArr here, just do simple iteration what is the point to iterate by chucks of 16 ?